src/referral_storage.cairo

```diff
@@ -0,0 +1,110 @@
+use starknet::{ContractAddress,ClassHash};
+use zeroable::Zeroable;
+
+
+#[starknet::interface]
+trait IReferralStorage<TContractState> {
+    fn get_trader_referral_code(
+         self: @TContractState,
+        _account: ContractAddress,
+    ) -> ContractAddress;
+
+    fn set_trader_referral_code(
+        ref self: TContractState,
+        _code: ContractAddress,
+    );
+
+    fn upgrade(ref self: TContractState, new_class_hash: ClassHash);
+}
+
+
+#[starknet::contract]
+mod ReferralStorage {
+    use starknet::ContractAddress;
+    use starknet::get_caller_address;
+    use zeroable::Zeroable;
+    use starknet::ClassHash;
+    use openzeppelin::access::ownable::OwnableComponent;
+    use openzeppelin::upgrades::UpgradeableComponent;
+    use openzeppelin::upgrades::interface::IUpgradeable;
+
+
+    component!(path: UpgradeableComponent, storage: upgradeable_storage, event: UpgradeableEvent);
+    component!(path: OwnableComponent, storage: ownable, event: OwnableEvent);
+
+    #[abi(embed_v0)]
+    /// Ownable
+    impl OwnableImpl = OwnableComponent::OwnableImpl<ContractState>;
+    impl OwnableInternalImpl = OwnableComponent::InternalImpl<ContractState>;
+
+    /// Upgradeable
+    impl UpgradeableInternalImpl = UpgradeableComponent::InternalImpl<ContractState>;
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        SetTraderReferralCode: SetTraderReferralCode,
```

Comment:
since the code is same as referrer contract address, maybe it's better to replace mentions of code to 'referrer'. For ex- change this event to 'SetTraderReferrer'. 

By using 'code', it seems like, each trader have a code, which they can share with others to earn referral bonus.

Reply:
changed it to SetReferrer